core/app/components/forms/elements/TextElement.tsx

```diff
@@ -27,6 +27,13 @@ export const TextElement = ({ label, name, ...rest }: ElementProps & InputProps)
 								{...fieldRest}
 								{...rest}
 								disabled={!isEnabled}
+								onChange={(e) => {
+									field.onChange(
+										rest.type === "number"
+											? Number(e.target.value)
+											: e.target.value
+									);
+								}}
```

Comment:
This casts the value of the `type="number"` input to a string before emitting it to the field change handler

Reply:
What do you think about making this numeric cast functionality part of our `Input` component? I am working on something that needs it but outside of a FormElement

Reply:
Oh that is a nice find! Eric mentioned that this was more complicated than I thought in a slack message on Thursday so we were not going to bother with it. Sorry we didn't update you and ty for confirming!

Reply:
oh okay! no worries, I'm only poking around here because I thought I'd try to get this PR across the line so I can use it for https://github.com/pubpub/platform/issues/614 :)